test/connect.cpp

```diff
@@ -566,13 +566,9 @@ BOOST_AUTO_TEST_CASE( noclean ) {
                         break;
                     case 3:
                         MQTT_CHK("h_connack4");
-                        // If clean session is not provided, than there will be a session present
-                        // if there was ever a previous connection, even if clean session was provided
-                        // on the previous connection.
-                        // This is because MQTTv5 change the semantics of the flag to "clean start"
-                        // such that it only effects the start of the session.
-                        // Post Session cleanup is handled with a timer, not with the  clean session flag.
-                        BOOST_TEST(sp == true);
+                        // The previous connection is not set Session Expiry Interval.
+                        // That means session state is cleared on close.
+                        BOOST_TEST(sp == false);
```

Comment:
I thought that if session expiry interval was not set, that means indefinite lifetime of the session data?

Reply:
Nervermind. I looked it up.

> If the Session Expiry Interval is absent the value 0 is used. If it is set to 0, or is absent, the Session ends when the Network Connection is closed.

Reply:
Ok. no problem.